templates/common/infra/bicep/core/host/container-app.bicep

```diff
@@ -123,7 +124,10 @@ resource app 'Microsoft.App/containerApps@2023-05-02-preview' = {
         appProtocol: daprAppProtocol
         appPort: ingressEnabled ? targetPort : 0
       } : { enabled: false }
-      secrets: secrets
+      secrets: [for secret in items(secrets): {
```

Comment:
Hm, are you saying this doesn't work? It seems to work, so am confused.
But yes it'd be nice to describe the shape somehow, agreed.

Reply:
> are you saying this doesn't work?

No, not saying this doesn't work. But, in order to work, you need to coordinate what you set in `env` parameter and what you set in `secrets` parameter.
You need to set an env array like:
```
[
   {
        name: FOO
        secretRef: refToSecretFoo
   }
]
```

Then a secret parameter like:
```
{
refToSecretFoo: {
   name: nameForTheSecret
   value: theSecretValueForFOO
}
}
```

So, I think we can merge both, `env` and `secrets` into the same input parameter, which is secured and which we use to build the container apps secrets and the environment for the container

Reply:
Ahh okay yes I understand. I think there's a drawback there in terms of debuggability, since you can no longer see the non-secret environment variables in your deployment inputs, but maybe that's worth it to have them be connected?

Reply:
@vhvb1989 I'm not sure I am following what you are saying.  In this PR we are literally just setting the `secrets` element of the bicep from the input parameter...

Reply:
> In this PR we are literally just setting the secrets element of the bicep

@wbreza , the `secrets element` is highly coupled to the template's `env element`, in the bicep definition.

What I am saying is that, we should change the module from taking `secrets` and `env` to only take `env`. Then, a customer would set which env-vars are secrets, and the module would take care of setting the secret and linking the env-var to it (instead or customer taking care of that).

Reply:
The container app bicep was intentional on the choice to expose both secrets and non-secret environment variables and I don't think it is our place to mask this choice.  Secrets require a different level of storage and being explicit upfront on the module make this more clear an intentional.

Reply:
FWIW, I don't think the design "masks the choice" of secret variables, or necessary exposes the storage details. The difference between the two is largely syntactical from an API perspective:

```
env: [
  {
     name: FOO
     value: BAR
  }
  {
     name: FOO_SECRET
     secretRef: foo-secret
   }
],
secrets: {
 foo-secret: BAR_SECRET
}
```

versus:

```
env: [
  {
     name: FOO
     value: BAR
  }
  {
     name: FOO_SECRET
     value: BAR_SECRET
     secret: true
  }
]
```

That being said. It's largely a choice of the bicep module author on what we want to expose. If we want to expose the raw details, that's fine. It *could* be possible that the user wanted to create a secret variable that is referenced elsewhere than the running container application, in which it isn't quite representable in the latter model.

I think we should proceed with fixing the issue, as we do very much want to mark it with `@secure()`. I see the downside of the current approach (which is present in other library modules as well) is that we always expose "raw details about the underlying bicep resource", but we lose out on both documentation and actual implementation of the details, which doesn't make it that much easier to write the desired resource. We certainly accomplished the goal of providing smart defaults, and that is an OK goal to have achieved in my mind.

Reply:
> I think there's a drawback there in terms of debuggability, since you can no longer see the non-secret environment variables in your deployment inputs

@pamelafox FWIW, if the `secret` was a KeyVault reference and not a manual secret value, you also lose out debuggability if that matters to you.